### PR TITLE
cxx-qt-gen: move #[qinvokable] to be inside extern "RustQt"

### LINKED ISSUE #558: Move `#[qinvokable]` into an `extern "RustQt"` block/outside the bridge
Use a declarative approach as can be seen in #555

### PR DESCRIPTION
**Pull request by ahayzen-kdab**
Requires #580

Closes #558

### CODE REVIEW

File: book/src/concepts/inheritance.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ To support creating such subclasses directly from within Rust, CXX-Qt provides y
 To access the methods of a base class in Rust, use the `#[inherit]` macro.
 It can be placed in front of a function in a `extern "RustQt"` block in a `#[cxx_qt::bridge]`.
 
+TODO: all of these examples need sorting out
```
Comment (ahayzen-kdab) [inline book/src/concepts/inheritance.md:18]: book needs checking

Comment (ahayzen-kdab) [reply book/src/concepts/inheritance.md]: tidied it up, but probably makes sense to rewrite the book similar to CXX with `extern "C++Qt"` and `extern "RustQt"` topics etc.

Comment (ahayzen-kdab) [reply book/src/concepts/inheritance.md]: Split that into #582

**Review by Be-ing — APPROVED** [review]
This is looking better :)

**Pull request merged**